### PR TITLE
add routes: add/remove employee in dept with tests, test db passed

### PR DESCRIPTION
**Pull request by phoebeww**

### CODE REVIEW

**Review by tonyxyu — APPROVED** [review]
Looks good to me

File: 100of100Service/src/main/java/dev/coms4156/project/command/AddEmployeeToDeptCommand.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      throw new IllegalArgumentException("Invalid date format. Expected yyyy-MM-dd.", e);
+    }
+
+    // Create the new employee
+    Employee newEmployee = new Employee(department.getEmployees().size() + 1, name, parsedHireDate);
+
+    // Add the employee to the department and sync with the database
+    department.addEmployee(newEmployee);
+    dbFacade.updateDepartment(department); // Sync with database
+
```
Comment (Alex-XJK) [inline 100of100Service/src/main/java/dev/coms4156/project/command/AddEmployeeToDeptCommand.java:60]: The new Employee object also need to sync to the database

File: 100of100Service/src/main/java/dev/coms4156/project/command/AddEmployeeToDeptCommand.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    Date parsedHireDate;
+    try {
+      SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+      parsedHireDate = sdf.parse(hireDate);
+    } catch (ParseException e) {
+      throw new IllegalArgumentException("Invalid date format. Expected yyyy-MM-dd.", e);
+    }
+
+    // Create the new employee
+    Employee newEmployee = new Employee(department.getEmployees().size() + 1, name, parsedHireDate);
```
Comment (Alex-XJK) [inline 100of100Service/src/main/java/dev/coms4156/project/command/AddEmployeeToDeptCommand.java:55]: I would recommend letting the `dbFacade` handle the employee insertion matters.
```java
Employee dummyEmployee = new Employee(-1, name, parsedHireDate);
Employee newEmployee = dbFacade.createEmployee(dummyEmployee);
// then do something with newEmployee
```

File: 100of100Service/src/main/java/dev/coms4156/project/command/AddEmployeeToDeptCommand.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+ */
+public class AddEmployeeToDeptCommand implements Command {
+  private final int clientId;
+  private final int departmentId;
+  private final String name;
+  private final String hireDate;
+
+  /**
+   * Constructs a command to add an employee to a given department.
+   *
```
Comment (Alex-XJK) [inline 100of100Service/src/main/java/dev/coms4156/project/command/AddEmployeeToDeptCommand.java:22]: This is a great and powerful endpoint, but I may think the `AddEmployeeToDept` will be used to assign an **existing employee** to an **existing department**. And maybe another endpoint is called `CreateEmployee` to create an employee but only add it to the organization without assigning it to a department.

**Pull request merged**